### PR TITLE
sm64ex: Various Features

### PR DESCRIPTION
**Pull request by N00byKing**
This adds an option to randomize both the normal courses as well as the secret courses throughout the game.

~However, there's a problem: I keep getting fill errors for what should be totally fine seeds.
If someone could take a look that'd be great~

EDIT: Now includes logic fixes as well for Big Star Doors, Options for setting MIPS Costs

### CODE REVIEW

File: worlds/sm64ex/Options.py [hunk 1]
```diff
@@ -44,9 +44,12 @@ class ExtraStars(Range):
     range_end = 50
     default = 50
 
-class AreaRandomizer(Toggle):
-    """Randomize Entrances to Courses"""
-    display_name = "Course Randomizer"
+class AreaRandomizer(Choice):
+    """Randomize Entrances"""
+    display_name = "Entrance Randomizer"
+    option_Off = 0
```
Comment (Alchav) [inline worlds/sm64ex/Options.py:50]: need to add `alias_false = 0` so this option does not cause generation failure

**Review by Alchav — CHANGES_REQUESTED** [review]
I generated several times and the only failure I got was related to the code comment I left. If you can get one of these fill failures to occur can you provide a yaml and seed number?

**Comment by N00byKing** [comment]
Generation failure only occurs when using Course and Secret Randomization.
Try Seed `81022829027716158669`, [yaml attached](https://github.com/ArchipelagoMW/Archipelago/files/9130279/yaml.zip)

**Comment by Alchav** [comment]
Appears what happened here is it placed DDD at BITFS, where the access rule requires being able to reach DDD, so DDD is unreachable.

**Comment by N00byKing** [comment]
THANK YOU!
This was getting extremely frustrating, but you're right on the money about the issue and 380 generations later and no issues means this was hopefully the last thing.
Sometimes you really do need a fresh set of eyes to find an issue.
Fixed this, and cleaned it up a bit as well. Ready for review

**Review by Joethepic — APPROVED** [review]

**Comment by espeon65536** [comment]
Have now looked at and played a game with it, seems fine to merge.

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**